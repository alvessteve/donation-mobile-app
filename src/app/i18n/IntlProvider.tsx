import React, { ReactElement, ReactNode } from 'react';
import { FormattedMessage, IntlProvider as BaseIntlProvider, useIntl as useBaseIntl } from 'react-intl';
import { FormatXMLElementFn, PrimitiveType } from 'intl-messageformat';
import { englishTranslations, Translations, TranslationsKeys } from './Translations';

export const useIntl = () => {
  const intl = useBaseIntl();

  return {
    translation: (id: TranslationsKeys, values?: Record<string, string | ((str: string) => ReactElement)>): string => intl.formatMessage({ id }, values as Record<string, PrimitiveType | FormatXMLElementFn<string, string>>),
  };
};

export default function IntlProvider(
  { children, overriddenTranslations = {} }: { children: ReactElement, overriddenTranslations?: Partial<Translations> },
) {
  return (
    <BaseIntlProvider
      messages={{ ...englishTranslations, ...(overriddenTranslations as Partial<Translations>) }}
      locale="en"
      defaultLocale="en"
    >
      {children}
    </BaseIntlProvider>
  );
}

export function Translation({ id, values = {} }: { id: TranslationsKeys, values?: Record<string, string | FormatXMLElementFn<ReactNode, ReactNode>> }) {
  return <FormattedMessage id={id} values={values} />;
}
