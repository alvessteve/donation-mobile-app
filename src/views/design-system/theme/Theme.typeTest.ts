/* eslint-disable @typescript-eslint/no-unused-vars */
import {
  ColorValueRgbType, ComponentsWithVariantColorPalette, Theme, ThemeVariant,
} from './stylesheet';

// @ts-expect-error
const wrongThemeValue: Theme = { aColor: 'rgb(61,139,190)' };

// @ts-expect-error
const wrongColorFormatOnTheme: Theme = { title: 'rgb(61,139,190)' };

// @ts-expect-error
const wrongHexaColor: ColorValueRgbType = 'FFF';

// @ts-expect-error
const wrongVariantTheme: ThemeVariant = 'rgb(61,139,190)';

// @ts-expect-error
const wrongThemeVariant: ComponentsWithVariantColorPalette = { tertiary: 'rgb(61,139,190)' };

// @ts-expect-error
const wrongColorFormat: ComponentsWithVariantColorPalette = { primary: 'rgb(61,139,190)' };
