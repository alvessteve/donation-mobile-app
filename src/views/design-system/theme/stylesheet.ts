/**
 * NB: we could restrict the ColorValueRgb type with the NumberRange type which restricts the boundary to 255 by a tail recursive type
 * however, I fell into a TS2590: Expression produces a union type that is too complex to represent
 type RgbUpperLimit = 255;
 type Mapped<N extends number, Result extends Array<number> = []> = (Result['length'] extends N ? Result : Mapped<N, [...Result, Result['length']]>);
 type NumberRange = Mapped<RgbUpperLimit>[number];
 */
export type ColorValueRgbType = `rgb(${number},${number},${number})`;
type ColorNames = 'curiousBlue'
| 'nobel'
| 'viking'
| 'white'
| 'black'
| 'lightgray'
| 'mediumLightGray'
| 'bittersweet';
type ColorPaletteType = Record<ColorNames, ColorValueRgbType>;

type SizeKeys = 'extraSmall' | 'small' | 'medium' | 'large' | 'extraLarge';
type SizeType = Record<SizeKeys, number>;

export type FontKeys = 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'caption';
type FontType = Record<FontKeys, number>;

export type Variant = 'primary' | 'secondary';

type ComponentsTheme = typeof lightThemeComponents;

export type Theme = {
  components: ComponentsTheme,
  palette: ColorPaletteType,
  typography: {
    size: SizeType,
    fontSize: FontType
  }
};

export const colorPalette: ColorPaletteType = {
  curiousBlue: 'rgb(61,139,190)',
  nobel: 'rgb(155,155,155)',
  viking: 'rgb(75,192,185)',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  lightgray: 'rgb(211,211,211)',
  mediumLightGray: 'rgb(182,182,182)',
  bittersweet: 'rgb(250,102,94)',
};

// act as a source of truth for the type to not have to declare each elements
const lightThemeComponents = {
  title: colorPalette.curiousBlue,
  subTitle: colorPalette.nobel,
  background: colorPalette.white,
  defaultText: colorPalette.black,
  button: {
    primary: {
      background: {
        initial: colorPalette.white,
        pressed: colorPalette.viking,
      },
      label: colorPalette.viking,
    },
    secondary: {
      background: {
        initial: colorPalette.viking,
        pressed: colorPalette.white,
      },
      label: colorPalette.white,
    },
  },
};

export const size: SizeType = {
  extraSmall: 8,
  small: 12,
  medium: 16,
  large: 28,
  extraLarge: 41,
};

export const fontSize: FontType = {
  p: 14,
  h1: 25,
  h2: 16,
  h3: 18,
  h4: 16,
  caption: 12,
};

const lightTheme: Theme = {
  components: lightThemeComponents,
  palette: colorPalette,
  typography: {
    size,
    fontSize,
  },
};

export const themeValues = { ...lightTheme };
