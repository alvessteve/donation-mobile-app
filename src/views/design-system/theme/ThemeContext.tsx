import React, { createContext, ReactNode, useContext } from 'react';
import { Theme, themeValues } from './stylesheet';

const ThemeContext = createContext<Theme>({} as Theme);

export const useTheme = () => useContext(ThemeContext);

export function ThemeProvider(
  { children }: { children: ReactNode },
) {
  return <ThemeContext.Provider value={themeValues}>{children}</ThemeContext.Provider>;
}
