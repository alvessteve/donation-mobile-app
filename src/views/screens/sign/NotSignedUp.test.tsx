import React from 'react';
import renderer from 'react-test-renderer';
import NotSignedUpScreen from './NotSignedUp';
import { ThemeProvider } from '../../design-system/theme/ThemeContext';
import IntlProvider from '../../../app/i18n/IntlProvider';

describe('NotSignedUp Page', () => {
  describe('Rendering', () => {
    it('should display the page', () => {
      const tree = renderer
        .create(
          <ThemeProvider>
            <IntlProvider>
              <NotSignedUpScreen />
            </IntlProvider>
          </ThemeProvider>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
