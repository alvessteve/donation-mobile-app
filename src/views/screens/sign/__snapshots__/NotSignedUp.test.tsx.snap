// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotSignedUp Page Rendering should display the page 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "rgb(255,255,255)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgb(61,139,190)",
            "fontSize": 25,
          },
          Object {
            "margin": 30,
          },
        ]
      }
    >
      Help contribute to impactful organisations
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(155,155,155)",
            "fontSize": 16,
          },
          Object {
            "marginLeft": 30,
            "marginRight": 30,
          },
        ]
      }
    >
      Give to organizations and view all your favorited charities in one place.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(255,255,255)",
            "borderColor": "rgb(75,192,185)",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
          },
          Object {
            "marginLeft": 30,
            "marginRight": 30,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(155,155,155)",
              "fontSize": 16,
            },
            Object {
              "color": "rgb(75,192,185)",
            },
          ]
        }
      >
        Log in
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(75,192,185)",
            "borderColor": "rgb(255,255,255)",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
          },
          Object {
            "marginLeft": 30,
            "marginRight": 30,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(155,155,155)",
              "fontSize": 16,
            },
            Object {
              "color": "rgb(255,255,255)",
            },
          ]
        }
      >
        Sign up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
